### ISSUE TITLE
Elementwise comparison operations should return boolean FixedArray

### ISSUE THREAD

**Issue by schlady** [SimonDanisch/FixedSizeArrays.jl#63]
``` julia
julia> Vec(1,2) .< Vec(1,2)
FixedSizeArrays.Vec{2,Int64}((0,0))

julia> Vec(1,2) .== Vec(1,3)
FixedSizeArrays.Vec{2,Int64}((1,0))
```

whereas for Base vectors we get

``` julia
julia> [1,2] .< [1,2]
2-element BitArray{1}:
 false
 false

julia> [1,2] .== [1,3]
2-element BitArray{1}:
  true
 false
```

**Comment by c42f** [comment]
Hmm, in the implementation `map` is used to map the comparison operators over the FSA elements, and the result is returned in the same FSA type as the input.  The problem is more apparent with floats:

``` julia
julia> Vec(1.0,2.0) .< Vec(1.0,2.0)
FixedSizeArrays.Vec{2,Float64}((0.0,0.0))
```

All FSAs are fed to `promote()` before being mapped over by the binary operators, with the mapping returning the promoted type as per above.  This obviously isn't the right thing for comparison operators.

A possible approach would be to use `Base.promote_op` in defining the binary operators http://docs.julialang.org/en/release-0.4/devdocs/promote-op/#devdocs-promote-op
but it looks like `Base.promote_op` and Base.Functors are not sufficiently developed or stable to rely on.

A better alternative might be to remove the use of `promote` in defining the binary operators, and instead beef up `map` to correctly deduce the output type of the mapping operation.  A `promote_op`-like system may be required to actually get this working.

**Comment by c42f** [comment]
Fixed via #105
